base/deprecated.jl

```diff
@@ -58,24 +58,22 @@ end
 
 function depwarn(msg, funcsym)
     opts = JLOptions()
-    if opts.depwarn > 0
+    if opts.depwarn == 1  # raise a warning
         ln = Int(unsafe_load(cglobal(:jl_lineno, Cint)))
         fn = String(unsafe_load(cglobal(:jl_filename, Ptr{Cchar})))
         bt = backtrace()
         caller = firstcaller(bt, funcsym)
-        if opts.depwarn == 1 # raise a warning
-            warn(msg, once=(caller != C_NULL), key=caller, bt=bt,
-                 filename=fn, lineno=ln)
-        elseif opts.depwarn == 2 # raise an error
-            throw(ErrorException(msg))
-        end
+        warn(msg, once=(caller != C_NULL), key=caller, bt=bt, filename=fn, lineno=ln)
+    elseif opts.depwarn == 2  # raise an error
+        throw(ErrorException(msg))
     end
 end
 
 function firstcaller(bt::Array{Ptr{Void},1}, funcsym::Symbol)
     # Identify the calling line
     i = 1
-    while i <= length(bt)
+    len = length(bt)
+    while i <= len
```

Comment:
lengths of arrays aren't constant, although loading this value from the stack spill vs. loading it from bt shouldn't be different in performance